### PR TITLE
add support to cache video only for web

### PR DESCRIPTION
**Pull request by moonshells**
For web page, when `cache.url` is not available or bidder wants to offload cache to server side, Prebid Server should be able to cache video.

### CODE REVIEW

**Comment by moonshells** [comment]
#324

**Comment by moonshells** [comment]
I will add the check which bidders are willing to use the Prebid Server cache, then we can use `cache.url` and `cacheMarkup` at the same time for different bidders.

**Comment by moonshells** [comment]
@sandraleon we definitely need tests. To be honest, looks like there's no unit test for method `auction`. I guess this's because auction is the "root" method that combined with endpoint "/auction", maybe we can use integration test to verify the behavior of `auction`. For me I tested my change manually. If you or @dbemiller could help setup this kind of end to end automation testing, I'm definitely willing to add test for case `pbs_req.CacheMarkup == 2`. @dbemiller, please let us know your opinion.

**Comment by dbemiller** [comment]
@moonshells given that it's legacy code, I definitely wouldn't put in the effort to make `/auction` tests. It's just too much effort for too little payoff.

I'd recommend moving your code into a separate function and writing unit tests around just that function instead. Similar to how `sortBidsAddKeywordsMobile` did it.

**Comment by moonshells** [comment]
@sandraleon @dbemiller I have moved the code into a separate function and added unit tests, please review it and let me know.

File: pbs_light.go [hunk 1]
```diff
@@ -469,6 +474,35 @@ func (deps *auctionDeps) auction(w http.ResponseWriter, r *http.Request, _ httpr
 	mRequestTimer.UpdateSince(pbs_req.Start)
 }
 
+// cache rubicon video bids only for Web
```
[the remark this answers is not in the document]
Comment (moonshells) [reply pbs_light.go]: What I know is bidders other than rubicon are caching video bids in Prebid.js via `cache.url`, I'm not sure whether they are willing to offload cache processing to the server side.

[the remark this answers is not in the document]
Comment (dbemiller) [reply pbs_light.go]: From the ticket @bretg filed, it sounded like this was a general problem affecting all bidders... am I wrong there?

**Comment by dbemiller** [comment]
Just FYI: I'm on vacation for a while starting from midday today. Feel free to count me as one of your approvals if you drop the `&& bid.BidderCode == "rubicon"` clause from the diff.

**Comment by moonshells** [comment]
@dbemiller `&& bid.BidderCode == "rubicon"` is removed, before `cacheMarkup == 2` being deployed, we should have https://github.com/prebid/Prebid.js/pull/2101 merged so DFP knows the cache id generated by PBS cache.

**Comment by moonshells** [comment]
@mkendall07 we should assign videoCacheKey with cache_id if we want to use PBS cache

**Comment by bretg** [comment]
Confirmed with moonshells that this change can be merged without the related PBJS change.

Dave is out, but spoke with @mkendall07 who 'dismissed' Dave's review since this item was covered:

```
drop the && bid.BidderCode == "rubicon" clause
```

That means there are virtually reviews from both Dave and Sandra, so merging this PR.

**Pull request merged**